src/universal/modules/meeting/components/MeetingRoot.js

```diff
@@ -0,0 +1,97 @@
+import ProjectDeletedSubscription from 'universal/subscriptions/ProjectDeletedSubscription';
+import ProjectUpdatedSubscription from 'universal/subscriptions/ProjectUpdatedSubscription';
+import TeamMemberAddedSubscription from 'universal/subscriptions/TeamMemberAddedSubscription';
+import TeamMemberUpdatedSubscription from 'universal/subscriptions/TeamMemberUpdatedSubscription';
+import NewAuthTokenSubscription from 'universal/subscriptions/NewAuthTokenSubscription';
+import NotificationsAddedSubscription from 'universal/subscriptions/NotificationsAddedSubscription';
+import NotificationsClearedSubscription from 'universal/subscriptions/NotificationsClearedSubscription';
+
+const query = graphql`
+  query MeetingRootQuery($teamId: ID!) {
+    viewer {
+      ...MeetingContainer_viewer
+    }
+  }
+`;
+
+const subscriptions = [
+  AgendaItemAddedSubscription,
+  AgendaItemUpdatedSubscription,
+  AgendaItemRemovedSubscription,
+  MeetingUpdatedSubscription,
+  ProjectUpdatedSubscription,
+  ProjectCreatedSubscription,
+  ProjectDeletedSubscription,
+  TeamMemberAddedSubscription,
+  TeamMemberUpdatedSubscription,
+  NewAuthTokenSubscription,
+  NotificationsAddedSubscription,
+  NotificationsClearedSubscription
+];
+
+const MeetingRoot = ({atmosphere, dispatch, history, location, match}) => {
+  const {params: {localPhase, localPhaseItem, teamId}} = match;
+  return (
+    <QueryRenderer
+      // flush the meeting immediately so we can start a new one
+      cacheConfig={{ttl: 1000}}
+      environment={atmosphere}
+      query={query}
+      variables={{teamId}}
+      subscriptions={subscriptions}
+      subParams={{dispatch, history, location}}
+      render={({error, props: renderProps}) => {
+        const {userId} = atmosphere;
+        const myTeamMemberId = `${userId}::${teamId}`;
+        return (
+          <TransitionGroup appear component={null}>
+            {error && <ErrorComponent height={'14rem'} error={error} />}
+            {renderProps &&
+            <AnimatedFade key="1">
+              <MeetingContainer
+                localPhase={localPhase}
+                localPhaseItem={localPhaseItem && Number(localPhaseItem)}
+                teamId={teamId}
+                myTeamMemberId={myTeamMemberId}
+                userId={userId}
+                viewer={renderProps.viewer}
+              />
+            </AnimatedFade>
+            }
+            {!renderProps && !error &&
+            <AnimatedFade key="2" unmountOnExit exit={false}>
+              <LoadingView />
+            </AnimatedFade>
+            }
+          </TransitionGroup>
+        );
+      }}
+    />
+  );
+};
+
+MeetingRoot.propTypes = {
+  atmosphere: PropTypes.object.isRequired,
+  dispatch: PropTypes.func.isRequired,
+  history: PropTypes.object.isRequired,
+  location: PropTypes.object.isRequired,
+  match: PropTypes.object.isRequired
+};
+
+export default connect()(withRouter(withAtmosphere(MeetingRoot)));
```

Comment:
Rather than calling `react-redux`'s `connect` on a parent which itself doesn't need `dispatch`, can we let the children that need access to `dispatch` connect themselves?

Reply:
that's a good practice! In this case, MeetingRoot does need dispatch to feed to the `subParams` because a certain subscription needs it (NotificationsAdded). May be a code smell to do it like this, but I don't know how else to couple the subscription to the query (and unfortunately they need coupling since we need to flush the query if the sub dies).